### ISSUE TITLE
dataLabels do not work properly on scatter charts since 3.0

### ISSUE THREAD

**Issue by thasmo** [apexcharts/apexcharts.js#1522]
https://eft.netlify.app/

Some data points to do not show a label, as you can see - and labels are assigned

I've tried every minor version from 3.0 to 3.18 and although the behavior and output changed between these versions, in general, the result was wrong for every version.

In the latest version, 3.18, some labels are missing and all other labels are wrong - it seems there is an index issue.

With 3.16 there are still some labels missing but those who show a label, have the correct label at least.

**Comment by thasmo** [comment]
It seems that there was a chance in 3.17.0 that somehow reorders the series datapoints unintentionally.

**Comment by junedchhipa** [comment]
There was a new property `xaxis.sorted` introduced in 3.17.1

Can you please try
```javascript
xaxis: {
  sorted: false
}
```

and see if it fixes the issue?

**Comment by thasmo** [comment]
@junedchhipa Indeed, that fixes it. Should this be disabled by default to prevent such issues?

**Comment by junedchhipa** [comment]
Absolutely.
I will disable `xaxis.sorted` unless it works fine for all scenarios.
